src/gui/tray/activitydata.cpp

```diff
@@ -50,6 +50,21 @@ OCC::Activity Activity::fromActivityJson(const QJsonObject &json, const AccountP
 {
     const auto activityUser = json.value(QStringLiteral("user")).toString();
 
+    const auto stringToUrl = [&] (const QString link) {
+        QUrl url(link);
```

Comment:
At first I thought it made sense but this is supposed to be a always a server url, sometimes even just part of it. These are links to the event to open the calendar in the browser or when you get a notification 'App X has an update' and you click on it and it opens the browser to the app page in the server. If we are getting something malicious from the server then at this point the server is already 'taken'.

Reply:
I have seen that we are using QUrl::fromUserInput for instance in the account set up wizard url

Reply:
not being an expert in this topic, I still think we should try to handle all incoming data in messages received with precaution
it may be an attack, an error, something else
in any case we should make sure we cannot do bad thing within the desktop client if wrong data are received

Reply:
I have changed it.